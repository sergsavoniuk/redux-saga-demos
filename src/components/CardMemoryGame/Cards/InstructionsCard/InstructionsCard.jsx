import React from 'react';

import withMemo from 'utils/withMemo';
import { Card } from '../Cards.components';

function InstructionsCard({ name, isFlipped, onCardClick }) {
  const content = !isFlipped ? (
    <p>{name}</p>
  ) : (
    <>
      <h2>Instructions</h2>
      <p>Press [p] to pause, or [ESC] to abandon game.</p>
      <p>
        Flip is a timed card memory game. Click the green cards to see what
        symbol they uncover and try to find the matching symbol underneath the
        other cards.
      </p>
      <p>
        Uncover two matching symbols at once to eliminate them from the game.
      </p>
      <p>
        Eliminate all cards as fast as you can to win the game. Have fun
        FLIPing!
      </p>
    </>
  );

  return (
    <Card flipped={isFlipped} onClick={onCardClick}>
      {content}
    </Card>
  );
}

export default withMemo(InstructionsCard, ['isFlipped']);
