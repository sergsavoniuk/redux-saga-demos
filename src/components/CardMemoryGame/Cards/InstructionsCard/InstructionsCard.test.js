import React from 'react';
import { shallow } from 'enzyme';

import InstructionsCard from './InstructionsCard';

describe('Testing InstructionsCard component', () => {
  let props;

  beforeEach(() => {
    props = {
      name: 'I',
      isFlipped: false,
      onCardClick: jest.fn(),
    };
  });

  test('renders correctly unflipped card', () => {
    const wrapper = shallow(<InstructionsCard {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('renders correctly flipped card', () => {
    props.isFlipped = true;

    const wrapper = shallow(<InstructionsCard {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should flip card after a click', () => {
    const wrapper = shallow(<InstructionsCard {...props} />);

    wrapper.simulate('click');
    expect(props.onCardClick).toHaveBeenCalledTimes(1);
    wrapper.setProps({ isFlipped: true });
    expect(wrapper.contains(<h2>Instructions</h2>)).toEqual(true);

    wrapper.simulate('click');
    expect(props.onCardClick).toHaveBeenCalledTimes(2);
    wrapper.setProps({ isFlipped: false });
    expect(wrapper.text()).toMatch(props.name);
  });
});
