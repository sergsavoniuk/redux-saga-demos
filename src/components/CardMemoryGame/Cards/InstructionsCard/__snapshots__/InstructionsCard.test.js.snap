// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InstructionsCard component renders correctly flipped card 1`] = `
<styled.div
  flipped={true}
  onClick={[MockFunction]}
>
  <h2>
    Instructions
  </h2>
  <p>
    Press [p] to pause, or [ESC] to abandon game.
  </p>
  <p>
    Flip is a timed card memory game. Click the green cards to see what symbol they uncover and try to find the matching symbol underneath the other cards.
  </p>
  <p>
    Uncover two matching symbols at once to eliminate them from the game.
  </p>
  <p>
    Eliminate all cards as fast as you can to win the game. Have fun FLIPing!
  </p>
</styled.div>
`;

exports[`Testing InstructionsCard component renders correctly unflipped card 1`] = `
<styled.div
  flipped={false}
  onClick={[MockFunction]}
>
  <p>
    I
  </p>
</styled.div>
`;
